fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/directory/FDBDirectory.java

```diff
@@ -516,6 +523,20 @@ public byte[] readStoredFields(String segmentName, int docId) throws IOException
         return rawBytes;
     }
 
+    @Nonnull
+    public List<KeyValue> readAllStoredFields(String segmentName) {
```

Comment:
I half-expected this to be `Cursor`-based, rather than synchronously making a list of key values, from which a list of primary key byte arrays. Maybe there is no legitimate need for such an optimization. How many documents in a segment do we expect?